pkg/cloud/cloud.go

```diff
@@ -217,15 +218,21 @@ var _ Cloud = &cloud{}
 
 // NewCloud returns a new instance of AWS cloud
 // It panics if session is invalid
-func NewCloud(region string) (Cloud, error) {
-	return newEC2Cloud(region)
+func NewCloud(region string, awsAssumeRole string) (Cloud, error) {
+	return newEC2Cloud(region, awsAssumeRole)
 }
 
-func newEC2Cloud(region string) (Cloud, error) {
+func newEC2Cloud(region string, awsAssumeRole string) (Cloud, error) {
 	awsConfig := &aws.Config{
 		Region:                        aws.String(region),
 		CredentialsChainVerboseErrors: aws.Bool(true),
 	}
+	sess := session.Must(session.NewSession(awsConfig))
+
+	if awsAssumeRole != "" {
+		klog.V(5).Infof("Assuming role into ARN %s", awsAssumeRole)
+		awsConfig.Credentials = stscreds.NewCredentials(sess, awsAssumeRole)
```

Comment:
I think these creds will expire. Is there a way to use the provider https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/stscreds/#AssumeRoleProvider to ensure sdk autorefershes them

Reply:
Whoops, I misread the docs and thought this was the thing that does that.
Thank you. I'll adjust.

Reply:
No, re-reading again, I think this code is correct.

Using `NewCredentials` is how the example code says to do it:
```// Create the credentials from AssumeRoleProvider to assume the role
// referenced by the "myRoleARN" ARN.
creds := stscreds.NewCredentials(sess, "myRoleArn")```
https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/stscreds/

And you can see in the sdk code, `NewCredentials` uses that provider for you:
https://github.com/aws/aws-sdk-go/blob/v1.36.7/aws/credentials/stscreds/assume_role_provider.go#L257

Also empirically, my `aws-ebs-csi-driver controller` process running on this branch has been up since dec4 and can still attach/detach.

Reply:
Bump. I've rebased and squashed on top of master as well.

Reply:
I think you're right:

```
... The Credentials value will attempt to refresh the credentials using the provider when Credentials.Get is called, if the cached credentials are expiring. 
```